server/server_test.go

```diff
@@ -85,6 +86,89 @@ func TestGetFile(t *testing.T) {
 	}
 }
 
+func TestAddLargeFile(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	stream, err := client.AddFile(ctx)
+	if err != nil {
+		t.Fatalf("failed to AddLargeFile: %v", err)
+	}
+
+	file, err := os.Open("test-data/IMG_2293.jpeg")
+	if err != nil {
+		t.Fatalf("failed to open file: %v", err)
+	}
+	defer file.Close()
+
+	const BufferSize = 1024
+
+	buffer := make([]byte, BufferSize)
+
+	stream.Send(&pb.AddFileRequest{Payload: &pb.AddFileRequest_AddParams{AddParams: &pb.AddParams{}}})
+
+	for {
+		bytesread, err := file.Read(buffer)
+
+		if err != nil {
+			if err != io.EOF {
+				t.Fatalf("failed while reading file: %v", err)
+			}
+			break
+		}
+		stream.Send(&pb.AddFileRequest{Payload: &pb.AddFileRequest_Chunk{Chunk: buffer[:bytesread]}})
+	}
+
+	resp, err := stream.CloseAndRecv()
+	if err != nil {
+		t.Fatalf("failed to CloseAndRecv AddLargeFile: %v", err)
+	}
+	refLargeFile = resp.GetNode()
+}
+
+func TestGetLargeFile(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+	stream, err := client.GetFile(ctx, &pb.GetFileRequest{Cid: refLargeFile.Block.GetCid()})
+	if err != nil {
+		t.Fatalf("failed to GetFile: %v", err)
+	}
+
+	out, err := os.Create("/tmp/out.jpeg")
+	defer out.Close()
+	if err != nil {
+		t.Fatalf("failed to create out file: %v", err)
+	}
+
+	buffer := bytes.NewBuffer([]byte{})
+	for {
+		resp, err := stream.Recv()
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			t.Fatalf("failed to receive file chunk: %v", err)
+		}
+		_, err = out.Write(resp.GetChunk())
+		if err != nil {
+			t.Fatalf("failed to write chunk to file: %v", err)
+		}
+
+		_, err = buffer.Write(resp.GetChunk())
+		if err != nil {
+			t.Fatalf("failed to write chunk to buffer: %v", err)
+		}
+	}
+
+	out.Sync()
+
+	got := int32(len(buffer.Bytes()))
+	expected := refLargeFile.GetStat().GetCumulativeSize() - refLargeFile.GetStat().GetLinksSize()
+	if got != expected {
```

Comment:
I'm trying to compare the size of the data that was added to the size of the data that comes back from GetFile. I can't figure out exactly what properties of the file that was added (`refLargeFile`) should be used to make that comparison, so this fails. The numbers are slightly different.

Reply:
I'm writing the file I'm getting to disk as well and it's identical to the original file. So things seem to be working.